packages/workspace/src/workspace/state/in_mem.ts

```diff
@@ -148,19 +149,15 @@ export const buildInMemState = (loadData: () => Promise<StateData>, persistent =
     setAll: async (elements: ThenableIterable<Element>): Promise<void> => awu(elements).forEach(setElement),
     remove: removeId,
     isEmpty: async (): Promise<boolean> => (await stateData()).elements.isEmpty(),
-    getAccountsUpdateDates: async () => {
-      const stateDataVal = await awu((await stateData()).accountsUpdateDate.entries()).toArray()
-      return Object.fromEntries(stateDataVal.map(e => [e.key, e.value]))
-    },
-    existingAccounts: async (): Promise<string[]> => awu((await stateData()).accountsUpdateDate.keys()).toArray(),
+    existingAccounts: async (): Promise<string[]> => (await (await stateData()).accounts.get('account_names')) ?? [],
```

Comment:
this should use the same implementation as `updateAccounts`
it should also _account_ for the possibility that this is an old format cache

```suggestion
    const existingAccounts = async (): Promise<string[]> => {
      const data = await stateData()
      const accounts = await data.accounts.get('account_names')
      if (accounts !== undefined) {
        return accounts
      }
      // Fallback to old format state cache where this information was written in a different map
      return await awu(data.accountsUpdateDate.keys()).toArray() ?? []
    }
```

Reply:
Ack. Also, I see what you did there.